pkgs/development/guile-modules/guile-reader/default.nix

```diff
@@ -0,0 +1,49 @@
+{ stdenv, fetchurl, pkgconfig
+, lightning
+, gperf, guile, guile-lib, libffi }:
+
+with stdenv.lib;
+stdenv.mkDerivation rec {
+
+  name = "guile-reader-${version}";
+  version = "0.6.2";
+
+  src = fetchurl {
+    url = "http://download.savannah.nongnu.org/releases/guile-reader/${name}.tar.gz";
+    sha256 = "0592s2s8ampqmqwilc4fvcild6rb9gy79di6vxv5kcdmv23abkgx";
+  };
+
+  patches = [(
+     fetchurl {
+       name = "register.patch";
+       url = https://aur.archlinux.org/cgit/aur.git/tree/register.patch?h=guile-reader-git&id=52beb87d4f6ff4c59e6b301ac1fbd574e0d9eaf2;
+       sha256 = "1mlq1lciqxf130s8wmmxi88l0h0gvjmcrky413cv3py7q4hk382g";
+     }) ];
+
+  nativeBuildInputs = [ pkgconfig ];
+  buildInputs = [ gperf guile guile-lib libffi ];
+
+  GUILE_SITE="${guile-lib}/share/guile/site";
+
+  preConfigure = ''
+    configureFlags="$configureFlags --with-guilemoduledir=$out/share/guile/site"
```

Comment:
If it is make package, you can just use https://github.com/NixOS/nixpkgs/blob/19ef534fbb044f2b95e0043c6e0f8b582d94952b/pkgs/desktops/gnome-3/misc/gpaste/default.nix#L18